### ISSUE TITLE
Implement CLI functionality, supporting a config file

### ISSUE THREAD

**Issue by phillipluther** [marksmccann/node-sass-extra#4]
Add command line API to `node-sass-extra` functionality.

- Basic set of params/flags for use without a config file?
    * _ (default) file or glob pattern for specifying source
    * -o/--out-file for writing compiled CSS to a specified file/dir
    * -c/--config for specifying path to config file (if not `./.nodesassrc.js`)
- Handling of options from a config file and mapping them to the appropriate functionality in the core script

**Comment by phillipluther** [comment]
Took me a while to get there, but I think I'm with you. I was initially thinking of tackling the problem with `-o` by leaving it alone and creating another CLI param called `--output-file` - that would be easy enough to map back and handle. But it would require an additional param in `index.js` to say that `dest.css` is a dir, not a file ... since our determination of whether something is a dir depends on the extension.

Not complicated, but that got me spiraling down the question of, "Is it worth it?"

The big value-adds of our tool are promise support and globs. The existing CLI supports globs, and I'd argue promise support is moot on the CLI. Maybe augmenting the CLI isn't really that critical ... at least for initial release.

**Comment by marksmccann** [comment]
Are you suggesting we drop this feature for the initial release? I am good with that. I will work on wrapping up the last couple issues so we can get this puppy published.

**Comment by phillipluther** [comment]
I am, yes - I was going to continuing hacking at it just to experiment, but I think the benefits we've added to the Node API are worth a publish enough as-is. 

Usage/feedback on the Node-side enhancements being ported to the CLI could be valuable, too.

**Comment by marksmccann** [comment]
@phillipluther I have been thinking about the CLI a lot lately. I definitely want to do it. A node-sass CLI that has all the features of our node API would give the developer a lot of autonomy and flexibility. Even simple tasks like creating a compressed output with a .min suffix would be possible in a single line. However, in our effort to maintain the existing node-sass API, I think it has tied our hands regarding the CLI. I am starting to think that if we abandon that notion (for the CLI only) it would allow us to move forward and create the best CLI possible. I am not suggesting that we make our CLI dramatically different from the original, I just think we can take a few liberties; specifically regarding the input and output syntax.

What do you think? How far did you get with the CLI? I would like to take a stab at it, I could build off of what you started? Also, is it possible to create unit tests for a CLI tool? – that is where I got stuck before.

**Comment by phillipluther** [comment]
I can't really think of any drop-in replacements for Node CLI tools; that could be ignorance, but I googled a bit and didn't really come up with anything, either. That got me wondering if we should even be entertaining that idea.

The `node-sass` package makes `node-sass` available on the command line, either locally or globally. For projects using our module, we'd essentially be overwriting that command ... and something smells wrong about that. Like overwriting the `webpack` or `babel` command. I can play around with it to test, but I'm not even sure NPM will let you do that.

Since we depend on `node-sass`, a consumer will have that command available to them to use as normal. We could provide a `node-sass-extra` command that unties our hands and lets us provide whatever CLI functionality we want. 

As far as testing the CLI, it's a bit of a pain - more integration testing than unit testing, since we wanna make sure the command line API is doing the expected actions VS. making sure specific functions return expected results. I've had success using [`spawn-command`](https://www.npmjs.com/package/spawn-command). It gives you event hooks into the child process for `stdout`, errors, and when the script stops running and stuff.

I'm rusty on that but it's doable.

I didn't get very far with the CLI - a basic script and argument parsing, and mapping glob support ... which of course, `node-sass` does on its own. I think right around then is when we started to question it and I shelved it.

I'll commit that up and if it helps, cool ... and if not, you're not missing much.

I kinda like the one-line minification idea and output file name manipulation you called out, too. Especially if we're accepting patterns for output like you suggested a while ago. So big long rambly, "Yes! That seems like a good path forward." A unique command that doesn't step on `node-sass` with robust pattern support for managing output.

**Comment by marksmccann** [comment]
Agreed. Sounds great. I'll take a stab at it soon, we can move on from there.
